src/EditorFeatures/CSharpTest/Classification/SemanticClassifierTests.cs

```diff
@@ -3612,5 +3612,38 @@ void N()
                 TypeParameter("T"),
                 Keyword("notnull"));
         }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Classification)]
+        public async Task DiscardInAssignment()
+        {
+            await TestAsync(@"
+class X
+{
+    void N()
+    {
+        int.TryParse("""", out var _);
+    }
+}",
+            Method("TryParse"),
+            Static("TryParse"),
+            Keyword("var"),
+            Keyword("_"));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Classification)]
+        public async Task ShortDiscardInAssignment()
+        {
+            await TestAsync(@"
+class X
+{
+    void N()
+    {
+        int.TryParse("""", out _);
+    }
+}",
+            Method("TryParse"),
+            Static("TryParse"),
+            Keyword("_"));
+        }
     }
```

Comment:
tests with tuples?

Reply:
Should the "_" be highlighted for tuples? Currently they aren't and the original Issue only talked about arguments.

Reply:
> Should the "_" be highlighted for tuples? Currently they aren't and the original Issue only talked about arguments.

IMO, yes.  If we're going to classify discards this way, we should consistently do it everywhere they come up.

Reply:
Ok. In that case i could use the new implementation and make it deal with `DiscardPatternSyntax` and `DiscardDesignationSyntax` which, according to the Synatx.xml should be the only two types we actually need to deal with. 

Do we have some kind of syntax generator that i can use to generate test cases for most types of syntax constructs containing discards?

Reply:
> Ok. In that case i could use the new implementation and make it deal with DiscardPatternSyntax and DiscardDesignationSyntax which, according to the Synatx.xml should be the only two types we actually need to deal with.

> Do we have some kind of syntax generator that i can use to generate test cases for most types of syntax constructs containing discards?

Yes.  He's called @jcouv :)